src/FffGcodeWriter.h

```diff
@@ -60,33 +60,37 @@ class FffGcodeWriter : public SettingsMessenger, NoCopy
     std::ofstream output_file;
 
     /*!
-     * Whether the skirt or brim polygons have been processed into planned paths
-     * for each extruder train.
+     * For each raft/filler layer, the extruders to be used in that layer in the order in which they are going to be used.
+     * The first number is the first raft layer. Indexing is shifted compared to normal negative layer numbers for raft/filler layers.
      */
-    bool skirt_brim_is_processed[MAX_EXTRUDERS];
+    std::vector<std::vector<unsigned int>> extruder_order_per_layer_negative_layers;
 
-    std::vector<std::vector<unsigned int>> mesh_order_per_extruder; //!< For each extruder, the cyclic order of the meshes (the first element is not the starting element per se)
+    std::vector<std::vector<unsigned int>> extruder_order_per_layer; //!< For each layer, the extruders to be used in that layer in the order in which they are going to be used
+
+    std::vector<std::vector<unsigned int>> mesh_order_per_extruder; //!< For each extruder, the order of the meshes (first element is first mesh to be printed)
 
     /*!
-     * For each extruder whether priming has already been planned
+     * For each extruder on which layer the prime will be planned,
+     * or a large negative number if it's already planned outside of \ref FffGcodeWriter::processLayer
+     * 
+     * Depending on whether we need to prime on the first layer, or anywhere in the print,
+     * the layer numbers are all zero (or less in case of raft)
+     * or they are the first layer at which the extruder is needed
```

Comment:
Previously this was a boolean and (as far as I know) therefore not possible to prime on a different layer than the first. Is this a new feature that was added at the same time? Or was this something that was unlocked by the changes you made to process the layers in parallel?